unyt/array.py

```diff
@@ -482,6 +482,7 @@ def __new__(
         dtype=None,
         bypass_validation=False,
         input_units=None,
+        name="",
```

Comment:
Shouldn't this be initialised to `None` not an empty str (as empty strs are mutable?)

Reply:
I started with None. I'll take a look.